internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -83,8 +83,27 @@ type PreprovisioningImageReconciler struct {
 // +kubebuilder:rbac:groups=metal3.io,resources=preprovisioningimages/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=agent-install.openshift.io,resources=infraenvs,verbs=get;list;watch
 // +kubebuilder:rbac:groups=agent-install.openshift.io,resources=infraenvs/status,verbs=get
+// +kubebuilder:rbac:groups=agent-install.openshift.io,resources=agentserviceconfigs,verbs=get;list;watch
```

Comment:
What do you think about just setting a bool in the controller that says whether or not image service is disabled instead of querying for the AgentServiceConfig in this controller?

Reply:
good idea :)